<launch>
  <arg name="hand_side" default="right"/>
  <arg name="checks_to_run" default="[position_sensor_noise_check, monotonicity_check]"/>
  
  <arg name="fingers_to_test" default="[FF, MF, RF, LF, TH, WR]"/>
  <arg name="results_path" default="/home/user/"/>
  
  <arg name="hand_serial" default="2346"/>

  <arg name="test_date" default="2020-02-10"/>
  <arg name="test_time" default="15-53-32"/>

  <arg name="real_hand" default="true"/>
  <arg name="rosbag_path" default="/home/user/sr_hand_health_reports/$(arg hand_serial)/health_report_results_$(arg test_date)_$(arg test_time)"/>
  <arg name="rosbag_name" default="health_report_bag_file.bag"/>

  <node name="sr_hand_health_report_node" pkg="sr_hand_health_report" type="sr_hand_health_report_node.py" args="-hs $(arg hand_side)" output="screen">
    <rosparam param="checks_to_run" subst_value="True">$(arg checks_to_run)</rosparam>
    <rosparam param="fingers_to_test" subst_value="True">$(arg fingers_to_test)</rosparam>
    <rosparam param="results_path" subst_value="True">$(arg results_path)</rosparam>
    <rosparam param="real_hand" subst_value="True">$(arg real_hand)</rosparam>
    <rosparam param="rosbag_path" subst_value="True">$(arg rosbag_path)</rosparam>
    <rosparam param="rosbag_name" subst_value="True">$(arg rosbag_name)</rosparam>
    <rosparam param="hand_serial" subst_value="True">$(arg hand_serial)</rosparam>
  </node>
</launch>
